### DISCUSSION TITLE
Question about using from_networkx

### DISCUSSION THREAD

**Discussion by errhernandez** [pyg-team/pytorch_geometric#2150] [Q&A] [answered]
Hi! Sorry in advance if this simply betrays my lack of understanding: I am using utils.from_networkx to create a Data object. My networkx graph has four features per node, and one feature per edge, as I construct it like this:

clusterx = nx.Graph()
for n in range(N):
     x, y, z = pos[n] 
     if n % 3 == 0: 
        spec = 0
     else: 
        spec = 1
     clusterx.add_node( n, species = spec, x = x, y = y, z = z )

and something similar for the edges; then I convert to a data object like so:

cluster = from_networkx( clusterx )

Everything fine up to here, but then, when I interrogate cluster (a torch_geometric.Data object) about the number of node features I get 1, when I expected to get 4 (spec, x, y, z). Am I misunderstanding something here?

All the data seems to be there in cluster, because cluster.keys produces 'x', 'edge_index', 'y', etc, but I am surprised by the fact that num_node_features produces an unexpected result. 

Sorry if this is a trivial issue. Thanks

Eduardo

**Comment by rusty1s** [comment, marked as answer]
This is to be expected since `num_node_features` will return the node feature size of `data.x`. In general, we expect that node features are grouped inside `data.x` - in your case, that would mean grouping `species`, `x`, `y` and `z` into a feature matrix of shape `[num_nodes, 4]`:
```python
data.x = torch.stack([data.species, data.x, data.y, data.z], dim=-1)
```

**Reply by errhernandez** [reply]
I see. Thanks!
